src/test/TestUtils/Assertions/CommandResultAssertions.cs

```diff
@@ -21,114 +21,115 @@ public CommandResultAssertions(CommandResult commandResult)
+                .FailWith("Command did not output anything to stderr.{0}", GetDiagnosticsInfo());
             return new AndConstraint<CommandResultAssertions>(this);
         }
 
         public AndConstraint<CommandResultAssertions> HaveStdErrContaining(string pattern)
         {
             Execute.Assertion.ForCondition(_commandResult.StdErr.Contains(pattern))
-                .FailWith(AppendDiagnosticsTo($"The command error output did not contain expected result: {pattern}{Environment.NewLine}"));
+                .FailWith("The command error output did not contain expected result: {0}{1}", pattern, GetDiagnosticsInfo());
             return new AndConstraint<CommandResultAssertions>(this);
         }
 
         public AndConstraint<CommandResultAssertions> NotHaveStdErrContaining(string pattern)
         {
             Execute.Assertion.ForCondition(!_commandResult.StdErr.Contains(pattern))
-                .FailWith(AppendDiagnosticsTo($"The command error output contained a result it should not have contained: {pattern}{Environment.NewLine}"));
+                .FailWith("The command error output contained a result it should not have contained: {0}{1}", pattern, GetDiagnosticsInfo());
             return new AndConstraint<CommandResultAssertions>(this);
         }
 
         public AndConstraint<CommandResultAssertions> HaveStdErrMatching(string pattern, RegexOptions options = RegexOptions.None)
         {
             Execute.Assertion.ForCondition(Regex.Match(_commandResult.StdErr, pattern, options).Success)
-                .FailWith(AppendDiagnosticsTo($"Matching the command error output failed. Pattern: {pattern}{Environment.NewLine}"));
+                .FailWith("Matching the command error output failed. Pattern: {0}{1}", pattern, GetDiagnosticsInfo());
             return new AndConstraint<CommandResultAssertions>(this);
         }
 
         public AndConstraint<CommandResultAssertions> NotHaveStdOut()
         {
             Execute.Assertion.ForCondition(string.IsNullOrEmpty(_commandResult.StdOut))
-                .FailWith(AppendDiagnosticsTo($"Expected command to not output to stdout but it was not:"));
+                .FailWith("Expected command to not output to stdout but it was not:{0}", GetDiagnosticsInfo());
             return new AndConstraint<CommandResultAssertions>(this);
         }
 
         public AndConstraint<CommandResultAssertions> NotHaveStdErr()
         {
             Execute.Assertion.ForCondition(string.IsNullOrEmpty(_commandResult.StdErr))
-                .FailWith(AppendDiagnosticsTo("Expected command to not output to stderr but it was not:"));
+                .FailWith("Expected command to not output to stderr but it was not:{0}", GetDiagnosticsInfo());
             return new AndConstraint<CommandResultAssertions>(this);
         }
 
-        private string AppendDiagnosticsTo(string s)
+
+        private string GetDiagnosticsInfo()
```

Comment:
I'm not sure this pattern is better.  Having to add `{0}` to the end of each string above seems unnecessary.

Why was this changed?

Reply:
Because the fluentassetions library didn't like the `=` in the literal error message. It basically ended up throwing a exception coming from String.Format().

Fixing this also explains why certain (unexpected) failures would not show the stdout information (throwing the String.Format exception), making it difficult to determine the cause -- the stdout contains some literal values that ended up throwing an exception.

Reply:
fwiw it's possible there was something in stdout and not the test string's `=` that was causing the issue (both were previously concat'd into one string literal); I didn't debug the issue to that level. Fluent assertions has its own expansion logic, so I believe it was getting tripped up by special characters